### PR TITLE
Remove EndpointSelectorContext

### PR DESCRIPTION
**Pull request by rynowak**
Doing the dirty work.

### CODE REVIEW

File: src/Http/Routing/src/EndpointRoutingMiddleware.cs [hunk 1]
```diff
@@ -55,12 +56,10 @@ internal sealed class EndpointRoutingMiddleware
 
         public Task Invoke(HttpContext httpContext)
         {
-            var feature = new EndpointSelectorContext(httpContext);
-
             // There's already an endpoint, skip maching completely
-            if (feature.Endpoint != null)
+            if (httpContext.GetEndpoint() != null)
```
Comment (JamesNK) [inline src/Http/Routing/src/EndpointRoutingMiddleware.cs]: nit: Assign GetEndpoint to variable and reuse variable

**Review by JamesNK — APPROVED** [review]
I couldn't find any extra lines to complain about

:shipit:

File: src/Http/Routing/src/EndpointRoutingMiddleware.cs (hunk 1 above)
Comment (rynowak) [reply src/Http/Routing/src/EndpointRoutingMiddleware.cs]: note that I'm still fixing 1.2 more thigns here.

**Review by davidfowl — APPROVED** [review]
Thanks. We should look at maybe exposing endpoint as a property or moving the ext method to a better ns

**Pull request merged**